src/clients/klist/klist.c

```diff
@@ -662,25 +662,27 @@ static void
 show_credential(krb5_creds *cred)
 {
     krb5_error_code ret;
-    krb5_ticket *tkt;
-    char *name, *sname, *flags;
+    krb5_ticket *tkt = NULL;
+    char *name = NULL, *sname = NULL, *tktsname, *flags;
     int extra_field = 0, ccol = 0, i;
+    krb5_boolean is_config = krb5_is_config_principal(context, cred->server);
 
     ret = krb5_unparse_name(context, cred->client, &name);
     if (ret) {
         com_err(progname, ret, _("while unparsing client name"));
-        return;
+        goto cleanup;
```

Comment:
Technically, here we could just return.

Reply:
We could, but if another allocation was added before this one, the `return` would have to be changed to `goto cleanup`.  For this reason I tend to use `goto cleanup` for the first of several allocations.  (I still use `return` when validating arguments prior to any allocations.)